Carnation refactoring

anpolimus opened the pull request:
Items, fixed in this PR.
1. Theme compile script changed from grunt to webpack
2. Refactored matchHeight, fixed: https://github.com/ymcatwincities/openy/issues/1692
3. Limited Time Offer paragraph looks ugly - fixed
4. User login page should be centralized vertically.
5. Fixed bug with images on tablet at homepage
6. Restored ability to configure logo from theme settings, made website logo working in the similar way to Rose and Lily
7. Refactored js files - split into small libraries that would be easier to override
8. Styled 403 and 404 pages
9. Fixed breadcrumbs margins at user pages
10. Fixed: Admin menu is showed behind carnation main menu
11. Added more block inheritance where it is possible at templates
12. Fixed Featured highlight block styles problems
13. Fixed Problem with small banner and alerts
14. Fixed  Small   Banner   paragraph  in   the   header   section
15. Fixed: Simple Content paragraph in the header section is not displaying at all
16. Fixed: When I add banner and gallery to the header section I see breadcrumbstwo times
17. Fixed: No   padding   between   paragraphs,   on   the   screenshot   below   is   aFeatured Highlights + Gallery.
18. Fixed: Margin between gallery and footer
19. Fixed: Featured Highlights paragraph + sidebar
20. Fixed: Grid columns paragraph + fontawesome icons: icons should becolored in the block color.
21. Fixed: Secondary Description & sidebar in the header area does not look good.
22. Fixed: Upcoming events paragraph is completely broken - implemented this paragraph template from scratch.
23. Fixed: adjust   styles   to   the   branch   info   section   on   the   branch   page
24. Moved logic from js to preprocesshttps://github.com/jediandrey/openy/blob/carnation_refactoring/themes/openy_themes/openy_carnation/src/js/openy_carnation_banner.js#L16
25. Fixed: Arrows are absent at the slider on mobile.
26. Updated Readme according to new theme compilation process.



## Steps for review

- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer CI. <br/><img src="https://raw.githubusercontent.com/ymcatwincities/openy/8.x-1.x/.github/assets/ci-errors.png" width="200" alt="CI code sniffer errors">
- [ ] All tests are running and there are no failed tests reported by CI. <br/><img src="https://raw.githubusercontent.com/ymcatwincities/openy/8.x-1.x/.github/assets/behat.png" width="200" alt="Behat test results">
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/><img src="https://raw.githubusercontent.com/ymcatwincities/openy/8.x-1.x/.github/assets/steps-for-review.png" width="200" alt="Steps for review"/>
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/><img src="https://raw.githubusercontent.com/ymcatwincities/openy/8.x-1.x/.github/assets/drupalorg-email.png" width="200" alt="drupal.org email"/>
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

retif committed:
OS-552 centered vertical align for off canvas content, for example login page form is centered

anpolimus committed:
Merge pull request #27 from retif/carnation_refactoring

OS-552 centered vertical align for off canvas content, for example

retif committed:
Fixed bug with images on tablet at homepage

anpolimus committed:
Merge pull request #28 from retif/OS-554

Fixed bug with images on tablet at homepage

retif committed:
openy_carnation.js refactoring

retif committed:
theme settings camp section added from lily

retif committed:
fix for Admin menu that showed behind carnation main menu on mobiles

retif committed:
Update themes/openy_themes/openy_carnation/src/js/openy_carnation_toolbar.js

Co-Authored-By: Anatolii Poliakov <anpolimus@gmail.com>

anpolimus committed:
Merge pull request #30 from retif/OS-574

fix for Admin menu that showed behind carnation main menu on mobiles

retif committed:
dev

retif committed:
Removed unnecesary dependency between logo rendering template and drupal block and region.
Logo is a stationary templete which is not required to be moved through regions.

retif committed:
removed unused html, added site_name variable

retif committed:
updated variable name for backward compatibility

retif committed:
changed title for secondary_menu region to more meaningful "User menu"

retif committed:
changed title for top_menu region to more meaningful "Mobile User menu"

retif committed:
openy_carnation.theme refactoring

retif committed:
openy_carnation.theme refactoring

retif committed:
deleted unused and confusing template

retif committed:
mobile logo for camps, fix for url generating function

retif committed:
main page logo

retif committed:
fixes grunt indentation errors

  3:6   error  Expected indentation of 4 spaces but found 5  indentation
  4:8   error  Expected indentation of 6 spaces but found 7  indentation
  5:10  error  Expected indentation of 8 spaces but found 9  indentation
  6:10  error  Expected indentation of 8 spaces but found 9  indentation

retif committed:
fixed breadcrumbs positioning for .without-banner page

retif committed:
styled 403 and 404 pages, fixed breadcrumbs position on those pages in css

anpolimus committed:
Merge pull request #29 from retif/OS-558

openy_carnation.js refactoring

retif committed:
removed logo css from primary menu first item

retif committed:
added secondary logo for pages with white background menu

retif committed:
removed buggy padding

anpolimus committed:
Merge pull request #33 from retif/OS-573

removed buggy padding

retif committed:
Hide Main menu in the Microsites menu for Carnation now works

retif committed:
fix white background logo url

retif committed:
Update _breadcrumbs.scss

retif committed:
Update _microsites.scss

anpolimus committed:
Merge pull request #32 from retif/OS-572

Os 572

anpolimus committed:
Merge pull request #31 from retif/OS-555

Os 555

anpolimus committed:
Merge pull request #34 from retif/OS-570

Hide Main menu in the Microsites menu for Carnation now works

retif committed:
is_array check before count call to avoid notice

retif committed:
changed title color for branch info

retif committed:
Added Email output for Branch Page

retif committed:
better css for upcoming events paragraph

retif committed:
new title color for branch info in scss

anpolimus committed:
Merge pull request #35 from retif/OS-590

OS-590

retif committed:
removed views suggestions alter which works by default

anpolimus committed:
Merge pull request #37 from retif/OS-589

better css for upcoming events paragraph

retif committed:
OS-585 Margin between gallery and footer removed

retif committed:
OS-583 Breadcrumbs was inserted twice by this jquery. Now it inserted after last item.

retif committed:
os-581 added description rendering in small banner

retif committed:
os-581 css refactoring. moved paragraph banner related css styles to paragraphs/_banner.scss

retif committed:
os-581 fixed breadcrumbs position for small banner

retif committed:
OS-584 added bottom margin for paragraph--featured-highlights

anpolimus committed:
Merge pull request #42 from retif/OS-584

OS-584 added bottom margin for paragraph--featured-highlights

anpolimus committed:
Merge pull request #41 from retif/OS-581

Os 581

anpolimus committed:
Merge pull request #40 from retif/OS-583

OS-583 Breadcrumbs was inserted twice by this jquery. Now it inserted…

anpolimus committed:
Merge pull request #39 from retif/OS-585

OS-585 Margin between gallery and footer removed

retif committed:
OS-587 icons colored in the block color

anpolimus committed:
Merge pull request #38 from retif/OS-587

Os 587

retif committed:
OS-579 Featured highlight block styles problems

retif committed:
.banner--small fix too big padding

anpolimus committed:
Merge pull request #43 from retif/OS-579

OS-579 Featured highlight block styles problems

retif committed:
removed duplicate variable

retif committed:
reformat code in openy_carnation.theme

retif committed:
OS-627 body classes and logo rendering moved from js to templates preprocess

retif committed:
OS-627 updated mobile logo, moved inline css to css files

retif committed:
OS-557 Refactor matchAllHeight

anpolimus committed:
Merge pull request #44 from retif/OS-557

OS-557 Refactor matchAllHeight

retif committed:
OS-627 moved breadcrumbs rendering from js to templates

retif committed:
OS-627 moved css from style attribute to css file in page.html.twig

retif committed:
OS-627 variable rename

atiupa committed:
[OS-551] Convert front-end asset compiling engine down to a small NPM script

anpolimus committed:
Merge pull request #46 from retif/OS-627

OS-627 body classes and logo rendering moved from js to templates preprocess

anpolimus committed:
Minor code styles fixes

atiupa committed:
[OS-551] Convert front-end asset compiling engine down to a small NPM script

atiupa committed:
[OS-551] Convert front-end asset compiling engine down to a small NPM script

anpolimus committed:
Merge remote-tracking branch 'origin/carnation_refactoring' into carnation_refactoring

anpolimus committed:
Added dependency for carnation theme

retif committed:
OS-628 Arrows fixed for the slider on mobile.

retif committed:
OS-574 removed previous attempt of admin toolbar fix

anpolimus committed:
Merge pull request #47 from retif/OS-628

Os 628 Arrows fixed for the slider on mobile

retif committed:
OS-603 limited time paragraph

anpolimus committed:
Merge pull request #48 from retif/OS-603

OS-603 limited time paragraph

anpolimus committed:
Updated readme for carnation

anpolimus committed:
Deleted old grunt code

ghost commented:
[**DeepCode Report (#0df1cf)**](https://www.deepcode.ai/app/gh/ymcatwincities/openy/f5312c73fdaf08a91f3d79da784f668bf5119f5a/ymcatwincities/openy/0df1cf8cd5a37196c8da8cd301813b8d4f64344c/_/%2F/code/)
---
DeepCode analyzed this pull request.
There are 2 new info reports.

bjwilkins commented:
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

anpolimus committed:
Merge branch '8.x-2.x' into carnation_refactoring

podarok commented on themes/openy_themes/openy_carnation/README.md:
```diff
@@ -1,50 +1,22 @@
+Carnation uses Webpack compiler. If you want to make any changes in css
+or js, please install Node.js and follow next instructions.
 
-`brew install yarn`
```
could you keep yarn part?
It's a better alternative to npm, so I guess we should still support it in readme

anpolimus replied:
Updated Readme.

anpolimus committed:
Updated initial config for carnation theme

anpolimus committed:
Code standards

anpolimus committed:
Merge remote-tracking branch 'origin/carnation_refactoring' into carnation_refactoring

anpolimus committed:
Recompile css after merge

podarok commented on themes/openy_themes/openy_carnation/openy_carnation.theme:
```diff
@@ -265,4 +458,111 @@ function openy_carnation_form_system_theme_settings_alter(&$form, FormStateInter
+}
+
+/**
+ * @param $fid_array
```
let's have a bit better, human-readable description ))))

anpolimus replied:
Updated

podarok commented on themes/openy_themes/openy_carnation/openy_carnation.theme:
```diff
@@ -265,4 +458,111 @@ function openy_carnation_form_system_theme_settings_alter(&$form, FormStateInter
+}
+
+/**
+ * @param $fid_array
```
same here
try php sniffer with Drupal codestyle or use IDE for highlighting wrong docblocks

anpolimus replied:
Updated

podarok commented on themes/openy_themes/openy_carnation/package.json:
```diff
@@ -9,35 +9,48 @@
+    "eslint": "^6.6.0",
+    "eslint-loader": "^3.0.2",
+    "mini-css-extract-plugin": "^0.8.0",
+    "node-sass": "^4.13.0",
```
let's also add polyfill for somehow IE11 compatible result
see https://github.com/ymcatwincities/openy/pull/1782/files#diff-aaaa85ee255ae55783f9abea93dc0bf5R12

podarok commented on themes/openy_themes/openy_carnation/webpack.build.js:
```diff
@@ -0,0 +1,9 @@
+module.exports = merge(common, {
+  mode: 'production',
+  optimization: {
+    minimize: true,
```
because we are doing minification, let's ensure in openy_carnation.libraries all js files have `minified:true` for not facing with aggregation issues
see https://github.com/ymcatwincities/openy/blob/8.x-2.x/themes/openy_themes/openy_lily/openy_lily.libraries.yml#L13

anpolimus replied:
Hi. At your example only CDN-loaded libraries are marked as minified.
Why do we need to add minified: true to our theme libraries?

podarok replied:
Because you do minification for them, aren't you?

podarok replied:
what this `'minimize': true` setting drives?

anpolimus replied:
We are not including those scripts to webpack prepared scripts.
We assuming that Drupal aggregation and minification will do this

podarok replied:
still confusing

podarok reviewed (changes requested):
There is a file mode change from 644 to 755, let's avoid it
And some other issues
Thank you

podarok committed:
Merge branch '8.x-2.x' into carnation_refactoring

podarok commented:
retest this please

bjwilkins commented:
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build1893/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/1893/

bjwilkins commented:
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build1893/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build1893/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build1893/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build1893/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build1893/blogWCAG2AAhtmlcs.txt

anpolimus commented:
@podarok , please dont merge it.
We found some regression issues and need to fix them here.

podarok committed:
Merge branch '8.x-2.x' into carnation_refactoring

bjwilkins commented:
Open Y Complete profile installed at http://openy.cibox.tools/build1893/

retif committed:
fixed header banner bug

anpolimus committed:
Merge pull request #49 from retif/carnation_refactoring

fixed header banner bug

anpolimus committed:
Added git ignore, deleted commented code

anpolimus committed:
Added Yarn to the README

anpolimus committed:
Code standards

anpolimus committed:
Added polyfills to packages

retif committed:
OS-633 breadcramb refactoring

retif committed:
margin bottom 5 for layout, because footer was too close to content

anpolimus committed:
Merge pull request #50 from retif/OS-633

Os 633 Breadcrumbs issues

atiupa committed:
Carnation refactoring: added babel-polyfill, some other fixes

atiupa committed:
Merge

atiupa committed:
Add updated style.css

anpolimus committed:
Fixed typo

retif committed:
subprogram filter form updated

anpolimus committed:
Merge pull request #51 from retif/OS-633

subprogram filter form updated

atiupa committed:
Carnation refactoring: fixed css mapping

atiupa committed:
Merge

anpolimus committed:
Improved button for filters at programs pages

anpolimus committed:
Merge remote-tracking branch 'origin/carnation_refactoring' into carnation_refactoring

# Conflicts:
#	themes/openy_themes/openy_carnation/dist/css/style.css

atiupa committed:
Carnation refactoring: make css extract/map different to dev and build

anpolimus committed:
Merge remote-tracking branch 'origin/carnation_refactoring' into carnation_refactoring

# Conflicts:
#	themes/openy_themes/openy_carnation/dist/css/style.css

anpolimus committed:
Updated padding for apply button, rebuilt css

retif committed:
OS-582 OS-570 microsites in the header, affix logo, login page logo, simple content paragraph in the header fixed
OS-634 Main menu at iPad Pro fixed, breakpoints for mobile menu changed
OS-586 Sidebar should go after content in case of tablet

anpolimus committed:
Merge pull request #53 from retif/OS-634

OS-582 OS-570 OS-634 OS-586 microsites, simple content in the header, Main menu at iPad Pro, sidebar on tablets

retif committed:
OS-631 Amenties block icons are fixed, new paragraph type created "Branch Amenities with icons"

retif committed:
OS-631 dependency injection

anpolimus committed:
Merge pull request #54 from retif/OS-631

OS-631 Amenties block icons are fixed, new paragraph type created "Br…

anpolimus committed:
Refactor of the block

anpolimus committed:
Merge branch '8.x-2.x' into carnation_refactoring

podarok commented:
retest this please

podarok commented on modules/openy_features/openy_block/modules/openy_block_branch_amenities/config/install/core.entity_form_display.paragraph.branch_amenities_with_icons.default.yml:
```diff
@@ -0,0 +1,16 @@
+    - paragraphs.paragraphs_type.branch_amenities_with_icons
+id: paragraph.branch_amenities_with_icons.default
+targetEntityType: paragraph
+bundle: branch_amenities_with_icons
```
I can't find where this component being enabled during installation of OpenY
I bet this should go to Extended version of OpenY , somewhere in advanced editorial https://github.com/ymcatwincities/openy/blob/8.x-2.x/openy.packages.yml#L82

anpolimus replied:
Hi.
This is addition to existed feature.
Since, this block only added for Carnation, we just giving ability to use this paragraph but not adding it to demo content (bc at Rose at Lily it would not be styled).
During next phase we will style it for Rose and Lily and add this paragraph to the Demo content.

podarok commented on modules/openy_features/openy_block/modules/openy_block_branch_amenities/config/install/field.storage.paragraph.field_branch_amenties_with_icons.yml:
```diff
@@ -0,0 +1,18 @@
+cardinality: 1
+translatable: true
+indexes: {  }
+persist_with_no_fields: false
```
should be true

anpolimus replied:
Done

bjwilkins commented:
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build1905/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/1905/

podarok commented on modules/openy_features/openy_block/modules/openy_block_branch_amenities/config/install/paragraphs.paragraphs_type.branch_amenities_with_icons.yml:
```diff
@@ -0,0 +1,8 @@
+id: branch_amenities_with_icons
+label: 'Branch Amenities with Icons'
+icon_uuid: null
+description: ''
```
could you add a description for content managers, please

anpolimus replied:
Done

bjwilkins commented:
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build1905/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build1905/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build1905/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build1905/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build1905/blogWCAG2AAhtmlcs.txt

podarok commented on themes/openy_themes/openy_carnation/.gitignore:
```diff
@@ -0,0 +1 @@
+index.html
```
remove file with a name "1" below, please

anpolimus replied:
Done

podarok commented on themes/openy_themes/openy_carnation/src/scss/paragraphs/_amenities.scss:
```diff
@@ -139,3 +139,80 @@
+    max-width: none;
+
+    .label-for-field-location-amenities {
+      font-family: "Cachet", sans-serif;
```
Cachet, Verdana, sans-serif

anpolimus replied:
Updated.

podarok reviewed (changes requested):
Remove the file named `1` in carnation folder
Chmod 644 openy_carnation_landing.js
Enable branch_amenities_with_icons in Extended
and some other minor things

anpolimus committed:
Added description to Branch Amenities with Icons paragraph + field fix

anpolimus committed:
Deleted unneeded file

anpolimus committed:
Changed hard-coded font with variable.

anpolimus committed:
Deleted commented code + changed permissions

podarok committed:
Merge branch '8.x-2.x' into carnation_refactoring

podarok commented:
retest this please

bjwilkins commented:
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build1906/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/1906/

bjwilkins commented:
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build1906/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build1906/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build1906/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build1906/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build1906/blogWCAG2AAhtmlcs.txt

bjwilkins commented:
Open Y Complete profile installed at http://openy.cibox.tools/build1906/
